pkg/channel/jetstream/dispatcher/reconciler.go

```diff
@@ -86,6 +95,86 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, nc *v1alpha1.NatsJetStre
 	return r.reconcileSubscriberStatuses(ctx, nc)
 }
 
+func (r *Reconciler) reconcileOrphanedSubscriptions(ctx context.Context, nc *v1alpha1.NatsJetStreamChannel) error {
+	logger := logging.FromContext(ctx)
+	if r.skipOrphanedSubscriptions {
+		return nil
+	}
+	var allSubsInNs, err = r.msgingClient.MessagingV1().Subscriptions(nc.Namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		logger.Errorw("failed to get subs", zap.Error(err))
+		return err
+	}
+
+	allSubsInNsUids := sets.New[string]()
+	for _, s := range allSubsInNs.Items {
+		allSubsInNsUids.Insert(string(s.UID))
+	}
+
+	subsUids := sets.New[string]()
+	for _, s := range nc.Spec.Subscribers {
+		subsUids.Insert(string(s.UID))
+	}
+
+	orphanedSubs := subsUids.Difference(allSubsInNsUids)
```

Comment:
I think we should have some sort of check for the type of Channel the subscription is referencing, as there are scenarios where users have multiple types of Channels (e.g. In Memory + NATS)

My concern is that we would be deleting too many subscriptions in this scenario

Reply:
@Cali0707 , it is difference, it works the next way:
> Difference returns a set of objects that are not in s2. For example: s1 = {a1, a2, a3} s2 = {a1, a2, a4, a5} s1.Difference(s2) = {a3} s2.Difference(s1) = {a4, a5}  

s1 in our case is `Channel.Spec.Subscriptions` so there is no issue if we take all subs in a namespace because we are making difference of `UIDS`. And we do not delete anything, we only delete orphaned subscriptions from a `Channel.Spec.Subscriptions` and `Channel.Status.Subscriptions`.